mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -180,9 +221,9 @@ def _azure_upload_file(self, credentials, local_file, artifact_path):
                             "Failed to authorize request, possibly due to credential expiration."
                             " Refreshing credentials and trying again..."
                         )
-                        credentials = self._get_write_credentials(
-                            self.run_id, artifact_path
-                        ).credentials
+                        credentials = self._get_write_credential_infos(
+                            run_id=self.run_id, paths=[artifact_path]
+                        )[0]
```

Comment:
Upon 403 errors from the cloud storage provider (S3, WASBS, etc), we fetch new credentials for the file being uploaded; this case rarely occurs and applies to large files which must be uploaded in many chunks. Credentials are fetched for a single path.